packages/snaps-rpc-methods/src/permitted/requestPermissions.ts

```diff
@@ -0,0 +1,103 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type {
+  PermissionConstraint,
+  PermittedHandlerExport,
+  SubjectPermissions,
+} from '@metamask/permission-controller';
+import type {
+  RequestPermissionParams,
+  RequestPermissionsResult,
+} from '@metamask/snaps-sdk';
+import type { Snap, SnapPermissions } from '@metamask/snaps-utils';
+import type { JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+import deepEqual from 'fast-deep-equal';
+
+import type { MethodHooksObject } from '../utils';
+
+export type RequestPermissionsHooks = {
+  getSnap: () => Snap | undefined;
```

Comment:
`getSnap` for permitted RPC methods is currently not origin bound. So we may want to create a new hook for accessing this 🤔

Reply:
My plan was to add the origin to the hook here: https://github.com/MetaMask/metamask-extension/blob/e956826aabb2b17508d66690170fca5a9c99daa0/app/scripts/metamask-controller.js#L4867
When I was testing I used that approach. My intention was to add the origin here and then move that hook outside of the keyring fencing.

Do you think I should create new hook for this use case or use the existing one?

Reply:
Updated the hook name to `getOriginSnap`. Will add that one in the PR for release changes.